### DISCUSSION TITLE
❌ Background app refresh?

### DISCUSSION THREAD

**Discussion by drmrbrewer** [isontheline/pro.webssh.net#821] [General]
I notice that my terminal session will terminate if the app is in the background for a short time.  It's easy enough to reconnect, but then you start a fresh session, and lose whatever state you had before.

I notice that many/most other apps (including Termius) have an option the the system settings for enabling "Background app refresh".  Would this be a way to keep those terminal sessions open and active, even when in the background?

**Comment by isontheline** [comment]
Termius uses GPS to force the app to be always active, isn't it?

It's not battery efficient and not privacy by design.

I would like Apple to implement a definitive way for app background operations.

**Comment by drmrbrewer** [comment]
So what purpose does background app refresh serve?

And how does Termius use GPS?  Just a hack to tell the system that it needs ongoing activity, like a map app?

**Reply by isontheline** [reply]
Yes a hack like this because app refresh doesn't serve this purpose.

You may read Termius' App Store details : "this app can track your location even when you don't use the app"

**Comment by drmrbrewer** [comment]
So what does background app refresh enable the app to do, if it’s not to refresh the app in the background?

**Reply by isontheline** [reply]
Perhaps syncing with Termius' cloud?

But I don't know at all.
